File: golf-gambling-rn/src/types/score.ts
export interface Score {
  id: string;
  holeId: string;
  playerId: string;
  gameId?: string; // Denormalized for queries
  strokes: number; // 0-15
  handicap: number; // 0-2
  isUp?: boolean; // Per-player 2x multiplier (deprecated - use multiplier instead)
  isBurn?: boolean; // Per-player 3x multiplier (deprecated - use multiplier instead)
  multiplier?: number; // Multiplier value: 1 (none), 2, 3, 4, or 6
}

export interface ScoreData extends Omit<Score, 'id'> {}

export interface ScoreWithDetails extends Score {
  playerName?: string;
  netScore: number;
}

File: golf_baccarat_tracker/lib/models/score.dart
class Score {
  final String id;
  final String holeId;
  final String playerId;
  final int strokes;
  final int handicap;

  Score({
    required this.id,
    required this.holeId,
    required this.playerId,
    required this.strokes,
    required this.handicap,
  });

  int get netScore => strokes - handicap;

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'holeId': holeId,
      'playerId': playerId,
      'strokes': strokes,
      'handicap': handicap,
    };
  }

  factory Score.fromMap(Map<String, dynamic> map) {
    return Score(
      id: map['id'] ?? '',
      holeId: map['holeId'] ?? '',
      playerId: map['playerId'] ?? '',
      strokes: map['strokes'] ?? 0,
      handicap: map['handicap'] ?? 0,
    );
  }

  Score copyWith({
    String? id,
    String? holeId,
    String? playerId,
    int? strokes,
    int? handicap,
  }) {
    return Score(
      id: id ?? this.id,
      holeId: holeId ?? this.holeId,
      playerId: playerId ?? this.playerId,
      strokes: strokes ?? this.strokes,
      handicap: handicap ?? this.handicap,
    );
  }
}
